### PR TITLE
SSI Substantial Gainful Activity

### PR DESCRIPTION
**Pull request by CalebPena**
fixes: #3664

### CODE REVIEW

File: policyengine_us/variables/gov/ssa/ssi/eligibility/income/ssi_income_eligible.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from policyengine_us.model_api import *
+
+
+class ssi_income_eligible(Variable):
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/ssa/ssi/eligibility/income/ssi_income_eligible.py:4]: Let's flip it to indicate SGA, and also disambiguate from the separate income limit (not exactly a limit but the income level at which SSI would fall to zero through the phase-out). Also we'll include ssi in the name to distinguish from SSDI's definition, which applies to blind people.
```suggestion
class ssi_engaged_in_sga(Variable):
```

File: policyengine_us/variables/gov/ssa/ssi/eligibility/income/ssi_income_eligible.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    value_type = bool
+    entity = Person
+    label = "Income less than the SGA limit"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        income = person("ssi_earned_income", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/ssa/ssi/eligibility/income/ssi_income_eligible.py:11]: just a note for the future that we'll want to deduct impairment-related work expenses https://github.com/PolicyEngine/policyengine-us/issues/3669

no need to address here

File: policyengine_us/variables/gov/ssa/ssi/eligibility/income/ssi_income_eligible.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from policyengine_us.model_api import *
+
+
+class ssi_income_eligible(Variable):
+    value_type = bool
+    entity = Person
+    label = "Income less than the SGA limit"
+    definition_period = YEAR
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/ssa/ssi/eligibility/income/ssi_income_eligible.py:8]:
```suggestion
    definition_period = YEAR
    reference = "https://www.ssa.gov/OP_Home/cfr20/416/416-0971.htm"
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/ssa/ssi/uncapped_ssi.py [hunk 4]
```diff
@@ -12,6 +12,9 @@ class uncapped_ssi(Variable):
     def formula(person, period, parameters):
         amount = person("ssi_amount_if_eligible", period)
         meets_resource_test = person("meets_ssi_resource_test", period)
+        meets_income_test = person("ssi_income_eligible", period)
         eligible = person("is_ssi_eligible_individual", period)
```
[the remark this answers is not in the document]
Comment (CalebPena) [reply policyengine_us/variables/gov/ssa/ssi/uncapped_ssi.py:16]: Will I have to update all of the other eligibility variables like `is_ssi_eligible_spouse` and `is_ssi_ineligible_child`

[the remark this answers is not in the document]
Comment (MaxGhenis) [reply policyengine_us/variables/gov/ssa/ssi/uncapped_ssi.py:16]: Good point - the simplest will be instead adding it as a criterion to `is_ssi_disabled`. Cleaner later on might be adding an intermediate variable `is_ssi_eligible_person` which is just `is_ssi_aged_blind_disabled & ~ssi_engaged_in_sga` and then using that in lieu of `is_ssi_aged_blind_disabled`.

File: policyengine_us/variables/gov/ssa/ssi/eligibility/income/ssi_income_eligible.py (hunk 1 above)
Comment (MaxGhenis) [reply policyengine_us/variables/gov/ssa/ssi/eligibility/income/ssi_income_eligible.py]: not quite what i suggested. let's make this income > sga

File: policyengine_us/variables/gov/ssa/ssi/eligibility/status/is_ssi_disabled.py [hunk 5]
```diff
@@ -13,6 +13,6 @@ def formula(person, period, parameters):
         aged = person("is_ssi_aged", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/ssa/ssi/eligibility/status/is_ssi_disabled.py:13]:
```suggestion
        # Classify person as disabled for SSI if:
        # 1) They reported receiving SSI due to presumed disability status (being neither aged nor blind); or
        # 2) They are disabled.
        # Earnings in excess of the Substantial Gainful Activity threshold disqualify from either case.
        # First check case 1: SSI reported and neither aged nor blind.
        aged = person("is_ssi_aged", period)
```

File: policyengine_us/variables/gov/ssa/ssi/eligibility/status/is_ssi_disabled.py [hunk 6]
```diff
@@ -13,6 +13,6 @@ def formula(person, period, parameters):
...
         blind = person("is_blind", period)
         reported_receipt = person("ssi_reported", period) > 0
-        return (reported_receipt & ~aged & ~blind) | person(
-            "is_disabled", period
-        )
+        is_disabled = person("is_disabled", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/ssa/ssi/eligibility/status/is_ssi_disabled.py:16]:
```suggestion
        reported_disabled_ssi = reported_receipt & ~aged & ~blind
        is_disabled = person("is_disabled", period)
        engaged_in_ssa = person("ssi_engaged_in_sga", period)
        return (reported_disabled_ssi | is_disabled) & ~engaged_in_ssa
```

**Review by MaxGhenis — APPROVED** [review]
Could you just make these changes and check comment line length? Placing precedence on SGA over reported SSI receipt won't affect MFB but it'll help us model policy reforms and microsimulation.

I'll merge from there.

File: policyengine_us/variables/gov/ssa/ssi/eligibility/status/is_ssi_disabled.py [hunk 7]
```diff
@@ -10,9 +10,17 @@ class is_ssi_disabled(Variable):
...
         blind = person("is_blind", period)
         reported_receipt = person("ssi_reported", period) > 0
-        return (reported_receipt & ~aged & ~blind) | person(
-            "is_disabled", period
-        )
+        reported_disabled_ssi = reported_receipt & ~aged & ~blind
+        is_disabled = person("is_disabled", period)
+        engaged_in_ssa = person("ssi_engaged_in_sga", period)
+        return (reported_disabled_ssi | is_disabled) & ~engaged_in_ssa
+        in_sga = person("ssi_engaged_in_sga", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/ssa/ssi/eligibility/status/is_ssi_disabled.py:25]: remove this and next line

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**